xarray/core/dataset.py

```diff
@@ -283,6 +287,18 @@ def attrs(self, value):
         self._attrs = OrderedDict(value)
 
     @property
+    def encoding(self):
+        """Dictionary of global encoding attributes on this dataset
+        """
+        if self._encoding is None:
+            self._encoding = OrderedDict()
+        return self._encoding
+
+    @encoding.setter
+    def encoding(self, value):
+        self._encoding = OrderedDict(value)
+
```

Comment:
Why do you need this getter & setter? Why not just have `self.encoding = OrderedDict(encoding)` in the `__init__`? Because you need to coerce to Ordered?